src/python/pants/source/source_root.py

```diff
@@ -17,7 +19,10 @@ class SourceRootCategories:
   ALL = [UNKNOWN, SOURCE, TEST, THIRDPARTY]
 
 
-SourceRoot = namedtuple('_SourceRoot', ['path', 'langs', 'category'])
+class SourceRoot(datatype([('path', str), 'langs', ('category', str)])):
+  pass
+
+SourceRootsCollection = Collection.of(SourceRoot)
```

Comment:
So is the feedback here that we should generally prefer to create custom types that are wrappers around native python collections, rather than using the `Collection.of` construction?

Reply:
Yes. And the reason is that it allows you to name it such that is unambiguously a _particular_ set of SourceRoots (namely "all the ones that exist on disk").

Reply:
So, actually, this would also be fine:
```
class AllExistingSourceRoots(Collection.of(SourceRoot)): pass
```

Because unlike `AllExistingSourceRoots = Collection.of(SourceRoot)` it results in a unique type. This is basically the difference between a typealias and a trait impl in rust. And the `datatype`y one above would be like a rust newtype.